pypiserver/__init__.py

```diff
@@ -151,12 +168,23 @@ def _make_root(root):
 
     c = default_config()
 
-    root = local_conf.get("root")
-    if root:
-        c['root'] = [_make_root(x) for x in root.split("\n") if x.strip()]
-
+    upd_conf_with_list_item(c, 'root', local_conf, sep='\n', parse=_make_root)
     upd_conf_with_bool_item(c, 'redirect_to_fallback', local_conf)
+    upd_conf_with_str_item(c, 'fallback_url', local_conf)
+    upd_conf_with_list_item(c, 'authenticated', local_conf, sep=' ')
+    upd_conf_with_str_item(c, 'password_file', local_conf)
     upd_conf_with_bool_item(c, 'overwrite', local_conf)
+    upd_conf_with_str_item(c, 'hash_algo', local_conf)
+    upd_conf_with_int_item(c, 'verbosity', local_conf)
+    upd_conf_with_str_item(c, 'log_file', local_conf)
+    upd_conf_with_str_item(c, 'log_frmt', local_conf)
+    upd_conf_with_str_item(c, 'log_req_frmt', local_conf)
+    upd_conf_with_str_item(c, 'log_res_frmt', local_conf)
+    upd_conf_with_str_item(c, 'log_err_frmt', local_conf)
+    upd_conf_with_str_item(c, 'welcome_file', local_conf)
```

Comment:
It would tighten this up a bit and make adding new values easier to pull the item types out into a list and run the update for all of them, at least for strings, e.g.:

```python
str_items = [
    'fallback_url',
    'password_file',
    'hash_algo',
    'log_file',
    'log_frmt',
    'log_req_frmt',
    'log_res_frmt',
    'log_err_frmt',
    'welcome_file'
]

for str_item in str_items:
    upd_conf_with_str_item(c, str_item, local_conf)
````